File: cases/taint/scala/loop-carried-positive/LoopCarriedPositive.scala
package dataflowbench

object LoopCarriedPositive {
  def dfb_source(): Int = { // DFB-SOURCE: loop-carried-input
    1
  }

  def dfb_sink(value: Int): Unit = {} // DFB-SINK: loop-carried-sink

  def run(): Unit = {
    var value = dfb_source()
    var iteration = 0
    while (iteration < 3) {
      value = value + iteration // DFB-WITNESS: loop-carried-value
      iteration = iteration + 1
    }
    dfb_sink(value)
  }
}

File: cases/taint/rust/loop-carried-positive/loop_carried_positive.rs
fn dfb_source() -> i32 { // DFB-SOURCE: loop-carried-input
    1
}

fn dfb_sink(value: i32) {} // DFB-SINK: loop-carried-sink

fn run() {
    let mut value = dfb_source();
    for iteration in 0..3 {
        value = value + iteration; // DFB-WITNESS: loop-carried-value
    }
    dfb_sink(value);
}
